application/libraries/REST_Controller.php

```diff
@@ -653,6 +653,32 @@ protected function preflight_checks()
     }
 
     /**
+    * Define controller method
+    *
+    * @access private
+    * @method define_controller_method
+    * @param  string $object_called
+    * @return string rename controller method
+    */
+	private function define_controller_method($object_called)
+	{
+		$rename_method = $object_called.'_'.$this->request->method;
+		$method_naming_convention = $this->config->item('method_naming_convention');
+
+		switch ( strtolower(str_replace(['_','-'], '', $method_naming_convention)) ) {
```

Comment:
`strtolower(str_replace(['_','-'], '', $method_naming_convention)`should not be needed. You cant avoid all the possibilities what the user can configure wrong.
Write the possibilities only down in the `rest.php`.

Else you have to care about all the wrong configurations and check if you can auto correct it. That overloads the library. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)